src/main/kotlin/com/kotlindiscord/bot/extensions/CleanExtension.kt

```diff
@@ -0,0 +1,162 @@
+package com.kotlindiscord.bot.extensions
+
+import com.gitlab.kordlib.cache.api.query
+import com.gitlab.kordlib.common.entity.Snowflake
+import com.gitlab.kordlib.core.cache.data.MessageData
+import com.gitlab.kordlib.core.entity.Message
+import com.gitlab.kordlib.core.entity.User
+import com.gitlab.kordlib.core.entity.channel.Channel
+import com.gitlab.kordlib.core.entity.channel.GuildMessageChannel
+import com.kotlindiscord.bot.authorId
+import com.kotlindiscord.bot.authorIsBot
+import com.kotlindiscord.bot.config.config
+import com.kotlindiscord.bot.defaultCheck
+import com.kotlindiscord.bot.enums.Roles
+import com.kotlindiscord.kord.extensions.ExtensibleBot
+import com.kotlindiscord.kord.extensions.checks.hasRole
+import com.kotlindiscord.kord.extensions.extensions.Extension
+import kotlinx.coroutines.flow.toList
+import mu.KotlinLogging
+import java.time.Instant
+
+/** Maximum number of deleted files allowed without the force flag. **/
+const val MAX_DELETION_SIZE = 50
+
+/** Milliseconds offset of the since parameter to make sure to delete the command invocation. **/
+const val SINCE_OFFSET = 100L
+
+private val logger = KotlinLogging.logger {}
+
+/** Extension providing a bulk message deletion command for mods+. **/
+class CleanExtension(bot: ExtensibleBot) : Extension(bot) {
+    override val name = "clean"
+
+    override suspend fun setup() {
+        @Suppress("ConstructorParameterNaming")
+        data class CleanArguments(
+            val user: List<User>? = null,
+            val regex: List<Regex>? = null,
+            val `in`: List<Channel>? = null,
+            val since: Message? = null,
+            val botonly: Boolean = false,
+            val count: Int = -1,
+            val force: Boolean = false
+        )
+
+        command {
+            name = "clean"
+            description = """
+                Delete messages in bulks. Available flags :
+                [user] : Only messages from this/those user(s).
+                [regex] : Only messages matching this/those regex(es).
+                [in] : The channel(s) to clean in. Default to the current one.
+                [since] : The first message to clean, up to the current one.
+                [botonly] : If true, only messages from bots.
+                [count] : Number of messages to clean.
+                [force] : Force deletion of messages. (admins only)
+            """.trimIndent()
+            aliases = arrayOf("clear", "c")
+
+            check(::defaultCheck)
+            check(hasRole(config.getRole(Roles.MODERATOR)))
+            signature<CleanArguments>()
+            hidden = true
+
+            action {
+                with(parse<CleanArguments>()) {
+                    val cleanNotice =
+                        """
```

Comment:
These quotes should be on the same line as the assignment operator. Also, see the feedback for the previous raw string.

It also looks like you might be doing a little too much with the string interpolation, but this may be cleaner.